### ISSUE TITLE
Doubt on reproducibility

### ISSUE THREAD

**Issue by KleinYuan** [epiception/CalibNet#19]
I try to reproduce your work based on your codebase and found the following doubts:

- You are using IMG_HT = 375, IMG_WDT = 1242 for training, which is different size of testing datasets. How do you even handle that? It makes no sense to me.

- Where's the network graph json? Could you at least share that? Otherwise, the network architecture is invisible to us.

- The photometric loss is very suspicious since you are using a l2 loss, which I assume will have at least 10k scale loss numerical value, while you are using the 5e-4 learning rate. Would you share your training curve on tensorboard?

**Comment by epiception** [comment]
Hi Klein,

1. All frames used for training and validation are of the specified height and width. Testing (Validation) frames are generated from the same set of folders from the original Kitti raw data sequences.

2. I did not generate a network graph json at the time of training.

3. You're not wrong about the scale of the loss term. Both the terms are weighted with respect to each other to ensure relatively smooth training. I don't see how the loss term should make a difference in the matter (apart from training being slightly slower). I don't have the tensorboard files at the moment, and I have no access to GPUs at the moment (since I don't work at the lab anymore). If you are training purely based on the instructions in the repository and face any issues, I'll be happy to assist.

**Comment by KleinYuan** [comment]
@epiception thanks for the quick response. No offense. I am just curious about some missing details:

1. From your paper, I read those

```

To demonstrate the generalization capability of our architecture even with different camera intrinsic parameters, 
we also use different instances from the 30_09 driving sequences during testing, as shown in Figure 5.

```
And 30_09 is of different size. And you claimed that the model trained on 26_09 can be generalized to 30_09, right? Then, how did you handle the differences between sizes?

1.1 
It will be great if you can clarify the `testing` and `validation` datasets you are using. Validation data is involved in training to fine tune the hyper-params, which is different than testing data. Are you saying that you are using `26_09` data for both training and testing, or you use `26_09` for training, then `26_09` non-training data for validation, and `30_09` for testing?

2.  This is the missing file I am talking about: https://github.com/epiception/CalibNet/blob/master/code/config_res.py#L11. It will be great if you can provide this since you later on implementation all depends on the structure of json, e.g,: https://github.com/epiception/CalibNet/blob/master/code/common/resnet_depth_model.py#L38

**Comment by epiception** [comment]
1. For the purposes of the paper, apart from 30_09, the code may use the term "validation" more freely, but for all purposes that was largely a part of the test set. For 30_09, although I do not clearly remember at the moment and can confirm later, I may have either reshaped the image or reduced the padding at the time of loading to ensure the network can handle any changes in height and width. So regarding the question, yes, part of all sequences for training, and non-training data for testing. I just used validation in code to ensure that I could release test sequences with different extrinsics at a later time, but since I left the lab, have lost access. 

2. The files are in the ReadMe: https://github.com/epiception/CalibNet#resnet-18

**Comment by KleinYuan** [comment]
Still not quite clear from you 1 answer.
So for your reported MAE, are you getting the results from testing data generated with`26_09`sequence or `30_09`? @epiception

**Comment by epiception** [comment]
Without 30_09 as far as I remember, but I can confirm this at a later time.

**Comment by KleinYuan** [comment]
@epiception I appreciate the answer! More clear on my end now.

**Comment by epiception** [comment]
Cool! Glad I could help!

**Comment by KleinYuan** [comment]
@epiception sorry to interrupt.
One more question:
In paper, you claimed that your reported translation MSE is 0.41 degree and this is calculated from average (4.2, 1.6, 7.22). And this number is obtained by using ground truth rotation and only give random translation noise between (-0.2 cm , 0.2 cm) on validation datasets from `26_09` sequence right?

More specifically, I am referring to the following sentence:

```
For this, we only use EMD as the error metric. We report mean absolute error in translation: (X: 4.2 cm, Y: 1.6 cm, Z: 7.22 cm).
```

And in `Translations given CalibNet rotation estimates` part, you mention a group of number `(X: 12.1 cm, Y: 3.49 cm, Z: 7.87 cm)`, are those number got with translation noise (-0.2, 0.2) or (-0.24, 0.24)? If it's (-0.24, 0.24), what is the performance of (-0.2, 0.2)?

Thanks
